include/no_os_spi.h

```diff
@@ -136,11 +155,33 @@ struct no_os_spi_init_param {
 	struct no_os_spi_desc *parent;
 };
 
+/**
+ * @struct no_os_spibus_desc
+ * @brief SPI bus descriptor
+*/
+struct no_os_spibus_desc {
```

Comment:
And if you do the no_os_spi.c modification that I proposed in another commit, you could then proceed to this one:

To make sure this desc is not modified by the user, you could declare here in the header:

```
struct no_os_spibus_desc;
```

And in the no_os_spi.c you could fully define the structure:
```
struct no_os_spibus_desc {
    ...
}

This way, you can only access the structure members by name in the no_os_spi.c